File: backend/app/schemas/plan.py
from pydantic import BaseModel
from app.schemas.vehicle import VehicleSummary, VehicleType


class PlanRequest(BaseModel):
    road_id: str
    lane_id: int


class PlanResponse(BaseModel):
    road_id: str
    road_name: str
    dump_trucks: int
    transfer_machines: int
    pavers: int
    rollers: int
    closure_vehicles: int
    suggested_vehicles: dict[VehicleType, list[VehicleSummary]]

File: frontend/src/types/plan.ts
import type { VehicleSummary, VehicleType } from './vehicle';

export interface PlanRequest {
  road_id: string;
  lane_id: number;
}

export interface PlanResponse {
  road_id: string;
  road_name: string;
  dump_trucks: number;
  transfer_machines: number;
  pavers: number;
  rollers: number;
  closure_vehicles: number;
  suggested_vehicles: Record<VehicleType, VehicleSummary[]>;
}
